diff --git a/vendor/magento/module-widget/Model/Template/Filter.php b/vendor/magento/module-widget/Model/Template/Filter.php
index c79334f67a9..87a4771f57a 100644
--- a/vendor/magento/module-widget/Model/Template/Filter.php
+++ b/vendor/magento/module-widget/Model/Template/Filter.php
@@ -5,6 +5,25 @@
  */
 namespace Magento\Widget\Model\Template;

+use Magento\Email\Model\Template\Css;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\State;
+use Magento\Framework\Css\PreProcessor\Adapter\CssInliner;
+use Magento\Framework\Escaper;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filter\VariableResolverInterface;
+use Magento\Framework\Stdlib\StringUtils;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Asset\Repository;
+use Magento\Framework\View\LayoutFactory;
+use Magento\Framework\View\LayoutInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Variable\Model\Source\Variables;
+use Magento\Variable\Model\VariableFactory;
+use Magento\Widget\Block\BlockInterface;
+use Magento\Widget\Model\Widget;
+use Psr\Log\LoggerInterface;
+
 /**
  * Template Filter Model
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
@@ -17,44 +36,54 @@ class Filter extends \Magento\Cms\Model\Template\Filter
     protected $_widgetResource;

     /**
-     * @var \Magento\Widget\Model\Widget
+     * @var Widget
      */
     protected $_widget;

     /**
-     * @param \Magento\Framework\Stdlib\StringUtils $string
-     * @param \Psr\Log\LoggerInterface $logger
-     * @param \Magento\Framework\Escaper $escaper
-     * @param \Magento\Framework\View\Asset\Repository $assetRepo
-     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
-     * @param \Magento\Variable\Model\VariableFactory $coreVariableFactory
-     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
-     * @param \Magento\Framework\View\LayoutInterface $layout
-     * @param \Magento\Framework\View\LayoutFactory $layoutFactory
-     * @param \Magento\Framework\App\State $appState
-     * @param \Magento\Framework\UrlInterface $urlModel
-     * @param \Pelago\Emogrifier $emogrifier
-     * @param \Magento\Variable\Model\Source\Variables $configVariables
+     * Filter constructor.
+     * @param StringUtils $string
+     * @param LoggerInterface $logger
+     * @param Escaper $escaper
+     * @param Repository $assetRepo
+     * @param ScopeConfigInterface $scopeConfig
+     * @param VariableFactory $coreVariableFactory
+     * @param StoreManagerInterface $storeManager
+     * @param LayoutInterface $layout
+     * @param LayoutFactory $layoutFactory
+     * @param State $appState
+     * @param UrlInterface $urlModel
+     * @param Variables $configVariables
+     * @param VariableResolverInterface $variableResolver
+     * @param Css\Processor $cssProcessor
+     * @param Filesystem $pubDirectory
      * @param \Magento\Widget\Model\ResourceModel\Widget $widgetResource
-     * @param \Magento\Widget\Model\Widget $widget
-     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     * @param Widget $widget
+     * @param CssInliner $cssInliner
+     * @param array $variables
+     * @param array $directiveProcessors
      */
     public function __construct(
-        \Magento\Framework\Stdlib\StringUtils $string,
-        \Psr\Log\LoggerInterface $logger,
-        \Magento\Framework\Escaper $escaper,
-        \Magento\Framework\View\Asset\Repository $assetRepo,
-        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
-        \Magento\Variable\Model\VariableFactory $coreVariableFactory,
-        \Magento\Store\Model\StoreManagerInterface $storeManager,
-        \Magento\Framework\View\LayoutInterface $layout,
-        \Magento\Framework\View\LayoutFactory $layoutFactory,
-        \Magento\Framework\App\State $appState,
-        \Magento\Framework\UrlInterface $urlModel,
-        \Pelago\Emogrifier $emogrifier,
-        \Magento\Variable\Model\Source\Variables $configVariables,
+        StringUtils $string,
+        LoggerInterface $logger,
+        Escaper $escaper,
+        Repository $assetRepo,
+        ScopeConfigInterface $scopeConfig,
+        VariableFactory $coreVariableFactory,
+        StoreManagerInterface $storeManager,
+        LayoutInterface $layout,
+        LayoutFactory $layoutFactory,
+        State $appState,
+        UrlInterface $urlModel,
+        Variables $configVariables,
+        VariableResolverInterface $variableResolver,
+        Css\Processor $cssProcessor,
+        Filesystem $pubDirectory,
         \Magento\Widget\Model\ResourceModel\Widget $widgetResource,
-        \Magento\Widget\Model\Widget $widget
+        Widget $widget,
+        CssInliner $cssInliner,
+        $variables = [],
+        array $directiveProcessors = []
     ) {
         $this->_widgetResource = $widgetResource;
         $this->_widget = $widget;
@@ -70,8 +99,13 @@ class Filter extends \Magento\Cms\Model\Template\Filter
             $layoutFactory,
             $appState,
             $urlModel,
-            $emogrifier,
-            $configVariables
+            $configVariables,
+            $variableResolver,
+            $cssProcessor,
+            $pubDirectory,
+            $cssInliner,
+            $variables,
+            $directiveProcessors
         );
     }

@@ -114,7 +148,7 @@ class Filter extends \Magento\Cms\Model\Template\Filter

         // define widget block and check the type is instance of Widget Interface
         $widget = $this->_layout->createBlock($type, $name, ['data' => $params]);
-        if (!$widget instanceof \Magento\Widget\Block\BlockInterface) {
+        if (!$widget instanceof BlockInterface) {
             return '';
         }

@@ -142,6 +176,6 @@ class Filter extends \Magento\Cms\Model\Template\Filter
     {
         $params = $this->getParameters(html_entity_decode($construction[2], ENT_QUOTES));
         return $this->_storeManager->getStore()
-            ->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_MEDIA) . $params['url'];
+                ->getBaseUrl(UrlInterface::URL_TYPE_MEDIA) . $params['url'];
     }
 }
