diff --git a/vendor/magento/module-wishlist/Controller/Shared/Cart.php b/vendor/magento/module-wishlist/Controller/Shared/Cart.php
index 38f10060297..c0a394ce9d7 100644
--- a/vendor/magento/module-wishlist/Controller/Shared/Cart.php
+++ b/vendor/magento/module-wishlist/Controller/Shared/Cart.php
@@ -3,13 +3,17 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 namespace Magento\Wishlist\Controller\Shared;
 
 use Magento\Catalog\Model\Product\Exception as ProductException;
 use Magento\Checkout\Helper\Cart as CartHelper;
 use Magento\Checkout\Model\Cart as CustomerCart;
+use Magento\Framework\App\Action\Action;
 use Magento\Framework\App\Action\Context as ActionContext;
-use Magento\Framework\App\Action\HttpGetActionInterface;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\Controller\Result\Redirect;
 use Magento\Framework\Controller\ResultFactory;
 use Magento\Framework\Escaper;
 use Magento\Framework\Exception\LocalizedException;
@@ -23,7 +27,7 @@ use Magento\Wishlist\Model\ResourceModel\Item\Option\Collection as OptionCollect
  *
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  */
-class Cart extends \Magento\Framework\App\Action\Action implements HttpGetActionInterface
+class Cart extends Action implements HttpPostActionInterface
 {
     /**
      * @var CustomerCart
@@ -80,7 +84,7 @@ class Cart extends \Magento\Framework\App\Action\Action implements HttpGetAction
      * If Product has required options - redirect
      * to product view page with message about needed defined required options
      *
-     * @return \Magento\Framework\Controller\Result\Redirect
+     * @return Redirect
      */
     public function execute()
     {
@@ -120,7 +124,7 @@ class Cart extends \Magento\Framework\App\Action\Action implements HttpGetAction
         } catch (\Exception $e) {
             $this->messageManager->addExceptionMessage($e, __('We can\'t add the item to the cart right now.'));
         }
-        /** @var \Magento\Framework\Controller\Result\Redirect $resultRedirect */
+        /** @var Redirect $resultRedirect */
         $resultRedirect = $this->resultFactory->create(ResultFactory::TYPE_REDIRECT);
         $resultRedirect->setUrl($redirectUrl);
         return $resultRedirect;
