diff --git a/vendor/magento/module-customer/Model/Address/CustomAttributesProcessor.php b/vendor/magento/module-customer/Model/Address/CustomAttributesProcessor.php
index d6e63e11ee4..0fd72a59189 100644
--- a/vendor/magento/module-customer/Model/Address/CustomAttributesProcessor.php
+++ b/vendor/magento/module-customer/Model/Address/CustomAttributesProcessor.php
@@ -71,7 +71,7 @@ class CustomAttributesProcessor
     {
         $attributeOptionLabels = [];
 
-        if (!empty($customAttribute['value'])) {
+        if (isset($customAttribute['value']) && $customAttribute['value'] != null) {
             $customAttributeValues = explode(',', $customAttribute['value']);
             $attributeOptions = $this->attributeOptionManager->getItems(
                 \Magento\Customer\Model\Indexer\Address\AttributeProvider::ENTITY,
