diff --git a/vendor/avalara/avatax-magento/Framework/Interaction/Rest/Tax/Cacheable.php b/vendor/avalara/avatax-magento/Framework/Interaction/Rest/Tax/Cacheable.php
index 34e805b4..c7e8d0ef 100644
--- a/vendor/avalara/avatax-magento/Framework/Interaction/Rest/Tax/Cacheable.php
+++ b/vendor/avalara/avatax-magento/Framework/Interaction/Rest/Tax/Cacheable.php
@@ -25,7 +25,7 @@ use Magento\Framework\App\CacheInterface;
 use Magento\Framework\Exception\LocalizedException;
 use ClassyLlama\AvaTax\Framework\Interaction\Rest\Tax\Result as TaxResult;
 use ClassyLlama\AvaTax\Exception\AvataxConnectionException;
-use Zend\Serializer\Adapter\PhpSerialize;
+use Laminas\Serializer\Adapter\PhpSerialize;
 
 /**
  * Class Cacheable
